ChustaSoft.Tools.Authorization/Configuration/AuthorizationSettingsBuilder.cs

```diff
@@ -56,11 +65,28 @@ public AuthorizationSettingsBuilder SetMinutesToExpire(int minutesToExpire)
             return this;
         }
 
-        public AuthorizationSettings Build() => _authorizationSettings;
+        public AuthorizationSettingsBuilder SetExternalAuthorization(IConfiguration configuration)
```

Comment:
IoC change:
Credentials from IConfiguration should be moved to ConfigurationHelper, we are adding the responsibility there to know about IConfiguration to the AuthorizationSettingsBuilder  when it is not required. Additionally, we are not allowing the tool to configure it from another mechanism different than appSettings.

The builder is thought to be prepared for configuring evth without Configuration

Reply:
My fault, I've moved the logic and now I think it matches the responsability of each class.